### PR TITLE
Add `hermitian()` to matrix gallery

### PR DESCRIPTION
**Pull request by ClaudiaComito**
## Due Diligence

- General:
    - [x]  **base branch** must be `main` for new features, latest release branch (e.g. `release/1.3.x`) for bug fixes
    - [x]  **title** of the PR is suitable to appear in the [Release Notes](https://github.com/helmholtz-analytics/heat/releases/latest)
- Implementation:
    - [x] unit tests: all split configurations tested
    - [x] unit tests: multiple dtypes tested
    - [x] documentation updated where needed

## Description

Creating a new PR to add tests for the Hermitian test matrix implemented in #1086.

Workflow: 
- merge #1086 into this PR ✅ 
- merge this PR into main

Issue/s resolved: #1073 

## Changes proposed:

-
-
-
-

#### Does this change modify the behaviour of other functions? If so, which?
no

### CODE REVIEW

File: heat/utils/data/matrixgallery.py [hunk 1]
```diff
@@ -1,153 +1,186 @@
...
+
+def hermitian(
+    n: int,
+    split: Union[None, int] = None,
+    device: Union[None, str, Device] = None,
+    comm: Union[None, Communication] = None,
+    dtype: Type[datatype] = core.complex64,
+) -> DNDarray:
+    """
+    Generates a Hermitian matrix of size `(n,n)`. A Hermitian matrix is a  complex square matrix that is equal to its conjugate transpose.
```
Comment (mrfh92) [inline heat/utils/data/matrixgallery.py:27]: With a small modification (devide by 1/size) we could add here that the matrices are sampled according to the "Marchenko-Pastur distribution" (https://en.wikipedia.org/wiki/Marchenko%E2%80%93Pastur_distribution)

File: heat/utils/data/matrixgallery.py [hunk 2]
```diff
@@ -1,153 +1,186 @@
...
+        The desired data-type for the array, defaults to ht.complex64.
+
+    References
+    ----------
+    [1] https://en.wikipedia.org/wiki/Hermitian_matrix
+    """
+    real_dtype = core.float32 if dtype is core.complex64 else core.float64
+    matrix = randn(n, n, dtype=real_dtype, split=split, device=device)
+    matrix += 1j * randn(n, n, dtype=real_dtype, split=split, device=device)
+    return matrix @ core.conj(matrix).T
```
Comment (mrfh92) [inline heat/utils/data/matrixgallery.py:49]: Two comments: 
* if we scale by 1/matrix.shape[0] the singular values are distributed according to Marchenko Pastur distribution (see above)... also this scaling might remove unwanted effects when choosing larger and larger matrices 
* The formula yields, of course, and they are even positive definite (with probability 1? - Im not an expert in this stuff...). However, doing the matrix-multiplication is quite expensive ... can we add something like an option "positive_definite" that triggers the formula used so far and use 
```
return matrix + core.conj(matrix).T.resplit(matrix.split) 
```
instead for the remaining cases?

**Review by mrfh92 — COMMENTED** [review]
only small comments on docs and formula used for generation of the random matrices. 
But in prinicple, this PR is ready for merging even without these changes. 

Thanks for the work :+1:

File: heat/utils/data/matrixgallery.py (hunk 2 above)
Comment (ClaudiaComito) [reply heat/utils/data/matrixgallery.py]: Thanks, done

**Comment by mrfh92** [comment]
@ClaudiaComito since I have done some changes here last week after my review it might be more useful if now you review again

**Review by mrfh92 — APPROVED** [review]
Looks fine to me, but since the last changes have been done by me, maybe somebody else should review too.

**Pull request merged**